src/explorer/collectionTreeItem.ts

```diff
@@ -24,103 +25,136 @@ class ShowMoreDocumentsTreeItem extends vscode.TreeItem {
 
   private _collectionName: string;
   private _databaseName: string;
   private _dataService: any;
-
-  isExpanded = false;
-
-  constructor(collectionName: string, databaseName: string, dataService: any) {
-    super(collectionName, vscode.TreeItemCollapsibleState.Collapsed);
-
-    this._collectionName = collectionName;
+  private _type: CollectionTypes;
+
+  isExpanded: boolean;
+
+  constructor(
+    collection: any,
+    databaseName: string,
+    dataService: any,
+    isExpanded: boolean,
+    existingChildrenCache: vscode.TreeItem[],
+    maxDocumentsToShow: number
+  ) {
+    super(collection.name, isExpanded ? vscode.TreeItemCollapsibleState.Expanded : vscode.TreeItemCollapsibleState.Collapsed);
+
+    this._collectionName = collection.name;
+    this._type = collection.type; // Type can be `collection` or `view`.
     this._databaseName = databaseName;
     this._dataService = dataService;
+
+    this.isExpanded = isExpanded;
+    this._childrenCache = existingChildrenCache;
+    this._maxDocumentsToShow = maxDocumentsToShow;
   }
 
   get tooltip(): string {
-    return this._collectionName;
+    return this._type === CollectionTypes.view
+      ? 'Read only view'
+      : this._collectionName;
   }
 
   getTreeItem(element: CollectionTreeItem): CollectionTreeItem {
     return element;
   }
 
   getChildren(): Thenable<any[]> {
-    if (this.isExpanded) {
-      if (this._childrenCacheIsUpToDate) {
-        return Promise.resolve(this._childrenCache);
-      }
-
-      return new Promise((resolve, reject) => {
-        const namespace = `${this._databaseName}.${this._collectionName}`;
-
-        this._dataService.find(
-          namespace,
-          {
-            /* No filter */
-          },
-          {
-            // We fetch 1 more than the max documents to show to see if
-            // there are more documents we aren't showing.
-            limit: 1 + this._maxDocumentsToShow
-          },
-          (err: any, documents: any[]) => {
-            if (err) {
-              return reject(`Unable to list documents: ${err}`);
-            }
-
-            this._childrenCacheIsUpToDate = true;
-
-            if (documents) {
-              this._childrenCache = documents.map((document, index) => {
+    if (!this.isExpanded) {
+      return Promise.resolve([]);
+    }
+
+    if (this._childrenCacheIsUpToDate) {
+      return Promise.resolve(this._childrenCache);
+    }
+
+    return new Promise((resolve, reject) => {
+      const namespace = `${this._databaseName}.${this._collectionName}`;
+
+      this._dataService.find(
+        namespace,
+        { /* No filter */ },
+        {
+          // We fetch 1 more than the max documents to show to see if
+          // there are more documents we aren't showing.
+          limit: 1 + this._maxDocumentsToShow
+        },
+        (err: Error, documents: []) => {
+          if (err) {
+            return reject(`Unable to list documents: ${err}`);
```

Comment:
Reject expects an `Error` instance and not just a string, no?

Reply:
Hmmm it looks like it works with both. That being said I think it's supposed to be `Error` instances - interesting that strings do the expected behavior in vscode though. I'll swap out all of our rejecting with strings to errors.